scripts/globals/mobskills/citadel_buster.lua

```diff
@@ -0,0 +1,86 @@
+---------------------------------------------
+-- Citadel Buster
+-- Deals extreme Light damage to players in an area of effect.
+-- Additional effect: Enmity reset
+-- Damage can be approximated based on Calculating Weapon Skill Damage as a magical WS with a level of 85, fTP of 6 and MAB of 4.0. Or, more simply:
+-- 2088/(1+MDB%) * (256-MDT)/256 (no day/weather bonus)
+-- 2608/MDB * (256-MDT)/256 (weather bonus)
+-- 2816/MDB * (256-MDT)/256 (day+weather bonus)
+---------------------------------------------
+require("scripts/globals/monstertpmoves")
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/weather")
+local ID = require("scripts/zones/Temenos/IDs")
+
+---------------------------------------------
+
+function onMobSkillCheck(target,mob,skill)
+    local phase = mob:getLocalVar("battlePhase")
+    if (phase == 4) then
+        mob:setLocalVar("citadelBuster", 1)
+        mob:SetMobAbilityEnabled(false)
+        mob:SetMagicCastingEnabled(false)
+        mob:SetAutoAttackEnabled(false)
+        mob:setMobMod(dsp.mobMod.DRAW_IN, 1)
+        local battlefield = mob:getBattlefield()
+        local players = battlefield:getPlayers()
+        for _, member in pairs(players) do
+            member:messageSpecial(ID.text.CITADEL_BASE)
```

Comment:
exactly zero of this belongs in the skillcheck, and should happen before the mob is even trying to activate the skill instead. Then when your countdown is over, force call the skill via the function we have for that purpose.

Reply:
I'm not sure where/how to move this without needing to manually set when to use the skill. It should be used randomly like any other tp move, but pulling these messages out of the skillcheck means I would need to manually call that at a certain hp percent or rely on random, which ends up just basically proccing over and over.

Reply:
I was about to explain the logic for timing it right, but then I realized the mixin for the mob 1hr/2hr moves can be hijacked for this to make ita  lot simpler. Unlock at 25% hp, allow it to get used more than once. mixin will handle the rest,

Reply:
1 min will talk in discord

Reply:
You could move it to the mobs script under onMobWeaponskill(mob, target, skill) I believe, then make it do the messages if skill = c. Buster

Reply:
The problem with that is onMobWeaponskill executes after the skill charging animation starts. The skill check it the only place with lua access between the ai picking which skill to use and the start of the charge animation.